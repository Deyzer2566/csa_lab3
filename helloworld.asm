b =_start
exint
message: string "hello, world!"
out_reg: word 0xfffffc
_start:
    or r0, message
    bl =find_len
    mov r1, r0
    and r0, 0
    or r0, message
    bl =print
    halt
find_len:
    mov r1, r0
    and r0, 0
loop:
    ldr r2, [r1]
    cmp r2, 0
    bxz lr
    add r0, 1
    add r1, 4
    b =loop
print:
    cmp r1, 0
    bxz lr
    ldr r2, [=out_reg]
    ldr r3, [r0]
    str r3, [r2]
    sub r1, 1
    add r0, 4
    b =print