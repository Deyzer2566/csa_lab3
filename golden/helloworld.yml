in_source: |-
  b =_start
  exint
  message: string "hello, world!"
  out_reg: word 0xfffffc
  _start:
      or r0, message
      bl =find_len
      mov r1, r0
      and r0, 0
      or r0, message
      bl =print
      halt
  find_len:
      mov r1, r0
      and r0, 0
  loop:
      ldr r2, [r1]
      cmp r2, 0
      bxz lr
      add r0, 1
      add r1, 4
      b =loop
  print:
      cmp r1, 0
      bxz lr
      ldr r2, [=out_reg]
      ldr r3, [r0]
      str r3, [r2]
      sub r1, 1
      add r0, 4
      b =print
in_stdin: |-

out_code: |-
  b'0\x00\x00@\xb8\x00\x00\x00\x00\x00\x00h\x00\x00\x00e\x00\x00\x00l\x00\x00\x00l\x00\x00\x00o\x00\x00\x00,\x00\x00\x00 \x00\x00\x00w\x00\x00\x00o\x00\x00\x00r\x00\x00\x00l\x00\x00\x00d\x00\x00\x00!\x00\x00\x00\x00\x00\xff\xff\xfcX\x00\x00\x088\x00\x00\x14\x08\x03\x00\x00P\x00\x00\x00X\x00\x00\x088\x00\x00$\xa0\x00\x00\x00\x08\x03\x00\x00P\x00\x00\x00\x10\x05\x10\x00\xb0\x04\x00\x00A\x96\x00\x00p\x00\x00\x01p\x02\x00\x040\x00\xff\xe8\xb0\x02\x00\x00A\x96\x00\x00\x10\x04\xff\xb4\x10\x07\x00\x00\x18\x07 \x00x\x02\x00\x01p\x00\x00\x040\x00\xff\xe0'
out_stdout: |
  hello, world!Выполнено тактов: 3817
